### PR TITLE
➖ remove broken `mdformat-pyproject`

### PR DESCRIPTION
**Pull request by miloth**
- Moved config to `.mdformat.toml` and removed `mdformat-pyproject` as the latter does not support all of the config. Solves #261.
- ~Simplified `poe` command now that the config works.~
- ~Explicitly stating that the dev environment should use Python 3.13, required by `mdformat`'s `exclude` config.~

### CODE REVIEW

File: pyproject.toml
Comment (jorenham) [inline pyproject.toml:109]: I usually use the lowest supported python version when developing, which can help catch bugs. And since `exclude` requires `python>=3.13`, the markdown files need to be explicitly passed for `mdformat` to work on `python<3.13`. I agree that it's not DRY and ugly, but in this case it's a necessary workaround. 

You could consider submitting a PR to `mdformat` that fixes it; I can't imagine it being all that difficult to implement.

**Comment by jorenham** [comment]
closely related to https://github.com/csala/mdformat-pyproject/issues/4

File: pyproject.toml
Comment (miloth) [reply pyproject.toml:109]: Another option here would be to remove the `mdformat` dependency (and its plugins) from the dev uv deps, keeping as a separate group:
```toml
[dependency-groups]
mdformat = ["mdformat", "plugins"]
```

Then, change the `poe` command to use a self-contained env made on purpose for the MD check:
```bash
uv run --isolated --group=mdformat --python=3.13 mdformat .
```

Comment (jorenham) [reply pyproject.toml:109]: Hmm, I think I prefer a list of explicit markdown files, as this will slow down stuff like pre-commit, which is bad for DX.

**Comment by jorenham** [comment]
Since it looks like `mkdformat-pyproject` isn't functional at the moment, I think that dropping it is indeed the best opption for now.

Could you rebase, so I can merge?

File: pyproject.toml
Comment (miloth) [reply pyproject.toml:109]: Do you want this reverted before merging?

File: CONTRIBUTING.md
Comment (jorenham) [inline CONTRIBUTING.md:193]:
```suggestion
> Python 3.13 is required to properly run `mdformat` (see the [documentation][mdformat-3.13]).
```

File: pyproject.toml
Comment (jorenham) [reply pyproject.toml:109]: Yea I think that would be best. And then maybe add a comment that says this is a workaround for the missing mdformat `exclude`with `python<3.13`.

File: CONTRIBUTING.md
Comment (jorenham) [inline CONTRIBUTING.md:1]: I suppose that these changes here also aren't needed if we keep the `python < 3.13` in place.

File: pyproject.toml
Comment (jorenham) [inline pyproject.toml:117]: I just raised https://github.com/hukkin/mdformat/issues/489. It could help to link to that here:

```suggestion
"""
# TODO: Remove once `--exclude` can be used on `python<3.13`, see
# https://github.com/hukkin/mdformat/issues/489
```

File: .mdformat.toml
Comment (jorenham) [inline .mdformat.toml:2]: won't `exclude` result in an error on `python<3.13`?

Comment (jorenham) [reply .mdformat.toml:2]: ... because if so, then this probably should be commented out until hukkin/mdformat#489 is implemented

**Comment by jorenham** [comment]
should be good to merge after the typo is fixed :)

**Review by jorenham — APPROVED** [review]

**Comment by jorenham** [comment]
Thanks @miloth

**Pull request merged**
